sdks/java/io/rabbitmq/src/test/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIOTest.java

```diff
@@ -203,25 +204,45 @@ private void doExchangeTest(ExchangeTestPlan testPlan, boolean simulateIncompati
     connectionFactory.setUri(uri);
     Connection connection = null;
     Channel channel = null;
+    Thread publisher = null;
+
+    // for debugging log messages
+    Map<String, String> testNameComponents = new HashMap<>();
+    testNameComponents.put("exchange", exchange);
+    testNameComponents.put("exchangeType", exchangeType);
+    testNameComponents.put("queue", Optional.ofNullable(read.queue()).orElse("<no queue>"));
+    final String testName = Joiner.on(", ").withKeyValueSeparator("=").join(testNameComponents);
 
     try {
       connection = connectionFactory.newConnection();
       channel = connection.createChannel();
       channel.exchangeDeclare(exchange, exchangeType);
       final Channel finalChannel = channel;
-      Thread publisher =
+      final String finalExchangeType = exchangeType;
+
+      publisher =
           new Thread(
               () -> {
                 try {
                   Thread.sleep(5000);
+                } catch (InterruptedException e) {
+                  // allow the test construct to cancel the thread
+                  LOG.info("Test {} interrupted before beginning publishing", testName);
```

Comment:
There was some exception noise in test output when tests fail-fast but the publisher thread continues attempting to publish on a closed channel. The Thread now checks for interrupt status and aborts if interrupted.